### PR TITLE
FaultInjection: Adds ApplyPercentage for ServerErrorRule

### PR DESCRIPTION
**Pull request by NaluTripician**
# Pull Request Template

## Description

Adds Application Percentage for ServerErrorRules. This allows users to specify how frequently server error rules are applied.  For instance, by setting this value to `.5` that rule will be applied 50% of the time. This value is 1, or 100%, by default. 
## Type of change

Please delete options that are not relevant.

- [] New feature (non-breaking change which adds functionality)

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/FaultInjectionServerErrorResultBuilder.cs [hunk 1]
```diff
@@ -65,6 +66,17 @@ public FaultInjectionServerErrorResultBuilder WithSuppressServiceRequest(bool su
             return this;
         }
 
+        public FaultInjectionServerErrorResultBuilder WithApplicationPercentage(double applicationPercentage)
+        {
+            if (applicationPercentage <= 0 || applicationPercentage > 1)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/FaultInjectionServerErrorResultBuilder.cs:71]: Is excluding 0 ZERO intentional?

File: Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/FaultInjectionServerErrorResult.cs [hunk 2]
```diff
@@ -14,19 +14,27 @@ public sealed class FaultInjectionServerErrorResult : IFaultInjectionResult
...
         /// <param name="times">Specifies the number of times a rule can be applied on a single operation.</param>
         /// <param name="delay">Specifies the injected delay for the server error.</param>
-        public FaultInjectionServerErrorResult(FaultInjectionServerErrorType serverErrorType, int times, TimeSpan delay, bool suppressServiceRequests)
+        /// <param name="applicationPercentage" Specifies the percentage of how many times the rule will be applied.</param>
+        public FaultInjectionServerErrorResult(
+            FaultInjectionServerErrorType serverErrorType, 
+            int times, 
+            TimeSpan delay, 
+            bool suppressServiceRequests,
+            double applicationPercentage = 1)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/FaultInjectionServerErrorResult.cs:31]: Name feels odd/un-intuitive suggestions? 

some possibilities: 
usagePercentage
applyPercentage

File: Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/FaultInjectionServerErrorResultBuilder.cs (hunk 1 above)
Comment (NaluTripician) [reply Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/FaultInjectionServerErrorResultBuilder.cs:73]: Yes, the value of 0 would mean that the rule would never be applied.

File: Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/implementataion/FaultInjectionServerErrorRule.cs [hunk 3]
```diff
@@ -19,6 +19,7 @@ internal class FaultInjectionServerErrorRule : IFaultInjectionRuleInternal
         private readonly FaultInjectionConnectionType connectionType; 
         private readonly FaultInjectionConditionInternal condition;
         private readonly FaultInjectionServerErrorResultInternal result;
+        private readonly Random random = new Random();
```
Comment (bartelink) [inline Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/implementataion/FaultInjectionServerErrorRule.cs:22]: if you're on net7 or later, just use Shared.Random

File: Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/FaultInjectionServerErrorResultBuilder.cs [hunk 4]
```diff
@@ -67,6 +68,17 @@ public FaultInjectionServerErrorResultBuilder WithSuppressServiceRequest(bool su
             return this;
         }
 
+        public FaultInjectionServerErrorResultBuilder WithApplyPercentage(double applyPercentage)
+        {
+            if (applyPercentage <= 0 || applyPercentage > 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(applyPercentage), "Argument 'applyPercentage' must be within the range (0, 1].");
```
Comment (kundadebdatta) [inline Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/src/FaultInjectionServerErrorResultBuilder.cs:75]: You can use string interpolation here to make it something like: 

`"Argument 'applyPercentage' must be within the range (0, 1]."` => `$"Argument '{nameOf(applyPercentage)}' must be within the range (0, 1]."`

**Review by kundadebdatta — APPROVED** [review]
LGTM!

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**